src/pages/docs/w3up-client.md

```diff
@@ -0,0 +1,123 @@
+
+To create a space using `w3up-client`, use the [`createSpace` client method][reference-w3up-client#createSpace]:
+
+```js
+const space = await client.createSpace('my-awesome-space')
+```
+
+The name parameter is optional. If provided, it will be stored in your client's local state store and can be used to provide a friendly name for user interfaces.
+
+After creating a [`Space`][reference-w3up-client#space], you'll need to register it with the w3up service before you can upload data.
+
+First, set the space as your "current" space using the [`setCurrentSpace` method][reference-w3up-client#setcurrentspace], passing in the DID of the `space` object you created above:
+
+```js
+await client.setCurrentSpace(space.did())
+```
+
+Next, call the [`registerSpace` method][reference-w3up-client#registerspace], passing in an email address to register as the primary contact for the space:
+
+```js
+try {
+  await client.registerSpace('zaphod@beeblebrox.galaxy')
+} catch (err) {
+  console.error('registration failed: ', err)
+}
+```
+
+Calling `registerSpace` will cause an email to be sent to the given address. Once a user clicks the confirmation link in the email, the `registerSpace` method will resolve. Make sure to check for errors, as `registerSpace` will fail if the email is not confirmed within the expiration timeout.
+
+Now that you've registered a space, you're ready to upload files!
+
+## Upload files
+
+Now that you've [created and registered a space](#create-space), you're ready to upload files to web3.storage!
+
+Call [`uploadFile`][reference-w3up-client#uploadfile] to upload a single file, or [`uploadDirectory`][reference-w3up-client#uploaddirectory] to upload multiple files.
+
+`uploadFile` expects a "Blob like" input, which can be a [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob) or [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File) when running in a browser. On node.js, see the [`filesFromPath` library](https://github.com/web3-storage/files-from-path), which can load compatible objects from the local filesystem.
+
+`uploadDirectory` requires `File`-like objects instead of `Blob`s, as the file's `name` property is used to build the directory hierarchy.
+
+You can control the directory layout and create nested directory structures by using `/` delimited paths in your filenames:
+
+```js
+const files = [
+  new File(['some-file-content'], 'readme.md'),
+  new File(['import foo'], 'src/main.py'),
+  new File([someBinaryData], 'images/example.png')
+]
+
+const directoryCid = await client.storeDirectory(files)
+```
+
+In the example above, `directoryCid` resolves to an IPFS directory with the following layout:
+
+```
+.
+├── images
+│   └── example.png
+├── readme.md
+└── src
+    └── main.py
+```
+## View your file on an IPFS gateway
```

Comment:
we have a similar section in cli. Should we just point both to a different page for IPFS Gateway?